mbuild/compound.py

```diff
@@ -1475,6 +1486,94 @@ def spin(self, theta, around):
+            else:
+                raise ValueError("Parameters align_this and with_this must be list-like of length 3"
+                                " containing numerical values.")
+            if np.isnan(aligner).any():
+                raise ValueError("Vector {} contains a NaN value. This function does not handle NaNs."
+                                 .format(aligner))
+        if anchor_pt is None:
+            anchor_pt = self.center
+        else:
+            try:
+                anchor_pt = np.array(anchor_pt, dtype=np.float64)
+            except ValueError:
+                raise ValueError("When describing the anchor_pt parameter as a set of cartesian"
+                                 " coordinates, it must be done by specifying a list-like of length 3"
+                                 " that contains exclusively numeric values. User passed {} of length {}"
+                                 " with non-numerical values.".format(type(anchor_pt), len(anchor_pt)))
+            try:
+                anchor_pt = anchor_pt.reshape((3,))
+            except ValueError:
+                raise ValueError("Parameter anchor_pt must be list-like of length 3. "
+                                 "User passed list-like of length {}.".format(len(anchor_pt)))
+            if np.isnan(anchor_pt).any():
+                raise ValueError("Vector {} contains a NaN value. This function does not handle NaNs."
+                                 .format(anchor_pt))
+
+        self._align_vector_with_vector(vec1=align_this, vec2=with_this, anchor_pt=anchor_pt)
+
+    def _align_vector_with_vector(self, vec1, vec2, anchor_pt):
+        """
+        This alignment technique assumes that all the input methods have been checked.
+        The function align_vector_with_vector() checks input and calls upon this to execute
+        the alignment. See def align_vector_with_vector() for more information.
+        """
+        self.translate(-anchor_pt)
+        goal = unit_vector(np.array(deepcopy(vec2)))
+        current = unit_vector(np.array(deepcopy(vec1)))
+        if np.allclose(goal,current, atol=1e-4):
+            warn("The vectors {} and {} are already aligned. "
+                 "No operation has been performed on the object."
+                 .format(vec1,vec2), UserWarning)
+            pass
+        elif np.allclose(goal, -1*current, atol=1e-4):
+            warn("The vectors {} and {} are parallel but point in opposite directions. No alignment performed."
+                 .format(vec1, vec2), UserWarning)
+            pass
+        else:
+            for passes in range(3):
+                orthag = np.cross(current, goal)
+                theta = angle(current, goal)
+                current = list(_rotate(coordinates=current, around=orthag, theta=theta))[0]
+                self.rotate(theta=theta, around=orthag)
+                if not np.allclose(goal,current, atol=1e-3):
+                    if passes > 5:
```

Comment:
What does this exactly do? It seems like `passes` will never be 5.

Reply:
it is very unlikely to even reach 3. This is in place to maximize accuracy and minimize floating pt error. When I was using hypothesis, I would occasionally get small errors when dealing with some of the more exhausting inputs. By using more passes, the desired outcome is virtually always achieved (I hate using always, but neither I or hypothesis could break it with the passes feature added in).

Reply:
I see what you're saying now. I'll fix that.